### ISSUE TITLE
Observe array of objects path

### ISSUE THREAD

**Issue by alnico2001** [BorisMoore/jsviews#459]
When a new item/object is added to an array, and paths to specific properties are being observed...there are observe inconsistencies...

Observe works for `number` and `ext` but does not work for newly added phones object as `phones` needs to be observed too...
`$.observe(person,"phones.[].number", "phones.[].ext", changeHandler);`

Therefore, this works for newly added phones object, but for `number` only....any additional paths using pattern: `phones^[]...` do not work (`ext`). 
`$.observe(person,"phones^[].number", "phones^[].ext", changeHandler);`

Here is one of your modified samples showing the issue: https://jsfiddle.net/alnico/8cgzwfnx/

To test: on load notice both `number` and `ext` are observed.
Click "Add phone" and change `number`, it is observed...now change `ext`...it is NOT observed.

Note: if the 'Add phone' button doesn't work, then copy code into your local env...no idea why it's not working in jsfiddle.

**Comment by BorisMoore** [comment]
Thanks for bringing this up, Brent. I'll look into it...

**Comment by BorisMoore** [comment]
Hi Brent, can you test whether this candidate fix works for you?

[jsviews.js.txt](https://github.com/BorisMoore/jsviews/files/15327891/jsviews.js.txt)

**Comment by alnico2001** [comment]
Thanks Boris...

The fix does not work in my app; it still only works on whichever path is first.
I see see of course that it works in the test file ;-)

This is my actual code using view models:
```
$.observe(appVm, 
   "locations()^[].Address()",
   "locations()^[].City()",
   "locations()^[].State()",
   "locations()^[].Zipcode()", 
    
mobileAddressChange)
```

**Comment by BorisMoore** [comment]
Ah - interesting. Would  you be able to provide me a small test example, with that pattern (or similar) observing view models? `"locations()^[].Address()"` (It will  save me a bit of  time :-) )

**Comment by alnico2001** [comment]
Here is an example I threw quickly together...it looks like the view model is causing the problem...

In this example file: only the first `"phones()^[]...."` works...even on initial rendering; never mind Add Phone...
```
$.observe(person,
   "phones()^[].number()",
   "phones()^[].ext()"
, changeHandler);
```
[test_jsviews_observVM.txt](https://github.com/BorisMoore/jsviews/files/15340702/test_jsviews_observVM.txt)

**Comment by BorisMoore** [comment]
Excellent. Thanks!,

**Comment by BorisMoore** [comment]
Hi Brent, here is a candidate, for you to test in your environment.

[jsviewsIssue459b.js.txt](https://github.com/BorisMoore/jsviews/files/15366713/jsviewsIssue459b.js.txt)

**Comment by alnico2001** [comment]
This fix works...Thank you Boris!

**Comment by BorisMoore** [comment]
Good.
These were in fact two different issues:
 `"phones^[].number", "phones^[].ext"`
and 
`"phones()^[]`...

Thanks for calling them out!

**Comment by BorisMoore** [comment]
Resolved, and included in commit v1.0.14
